crates/router/src/connector/bamboraapac.rs

```diff
@@ -167,6 +183,85 @@ impl
         types::PaymentsResponseData,
     > for Bamboraapac
 {
+    fn get_headers(
+        &self,
+        req: &types::SetupMandateRouterData,
+        connectors: &settings::Connectors,
+    ) -> CustomResult<Vec<(String, request::Maskable<String>)>, errors::ConnectorError> {
+        self.build_headers(req, connectors)
+    }
+
+    fn get_content_type(&self) -> &'static str {
+        self.common_get_content_type()
+    }
+
+    fn get_url(
+        &self,
+        _req: &types::SetupMandateRouterData,
+        connectors: &settings::Connectors,
+    ) -> CustomResult<String, errors::ConnectorError> {
+        Ok(format!("{}/sipp.asmx", self.base_url(connectors)))
+    }
+
+    fn get_request_body(
+        &self,
+        req: &types::SetupMandateRouterData,
+        _connectors: &settings::Connectors,
+    ) -> CustomResult<RequestContent, errors::ConnectorError> {
+        let connector_req = bamboraapac::get_setup_mandate_body(req)?;
+
+        Ok(RequestContent::RawBytes(connector_req))
```

Comment:
For longer term though, we may want to use a different variant specific for SOAP requests.